File: src/main/java/com/company/erp/features/chats/dto/CallParticipantDto.java
package com.company.erp.features.chats.dto;

import com.company.erp.features.chats.entity.ChatCallParticipant;
import com.company.erp.features.chats.entity.ParticipantStatus;

import java.time.Instant;

public record CallParticipantDto(
        Long userId,
        ParticipantStatus status,
        Instant joinedAt,
        Instant leftAt
) {
    public static CallParticipantDto from(ChatCallParticipant p) {
        return new CallParticipantDto(p.getUserId(), p.getStatus(), p.getJoinedAt(), p.getLeftAt());
    }
}

File: app/Features/Chat/Dto/CallParticipantDto.php
<?php

namespace App\Features\Chat\Dto;

use App\Features\Chat\Models\ChatCallParticipant;

/** Port of the Spring CallParticipantDto record. */
final class CallParticipantDto
{
    /**
     * @return array<string, mixed>
     */
    public static function from(ChatCallParticipant $p): array
    {
        return [
            'userId' => (int) $p->user_id,
            'status' => $p->status,
            'joinedAt' => $p->joined_at ? $p->joined_at->utc()->format('Y-m-d\TH:i:s\Z') : null,
            'leftAt' => $p->left_at ? $p->left_at->utc()->format('Y-m-d\TH:i:s\Z') : null,
        ];
    }
}
